packages/vite/src/node/plugins/html.ts

```diff
@@ -116,14 +116,261 @@ export function addToHTMLProxyTransformResult(
   htmlProxyResult.set(hash, code)
 }
 
-// this extends the config in @vue/compiler-sfc with <link href>
-export const assetAttrsConfig: Record<string, string[]> = {
-  link: ['href'],
-  video: ['src', 'poster'],
-  source: ['src', 'srcset'],
-  img: ['src', 'srcset'],
-  image: ['xlink:href', 'href'],
-  use: ['xlink:href', 'href']
+export interface HtmlAssetSource {
+  tag: string
+  attributes: string[]
+  type: 'src' | 'srcset'
+  filter?: (attribute: string, value: string, node: ElementNode) => boolean
+}
+
+type AttributeNodeContext = {
+  context: {
+    sourceType: 'src' | 'srcset'
+  }
+}
+
+export const defaultHtmlAssetSources: HtmlAssetSource[] = [
+  {
+    tag: 'audio',
+    type: 'src',
+    attributes: ['src']
+  },
+  {
+    tag: 'embed',
+    type: 'src',
+    attributes: ['src']
+  },
+  {
+    tag: 'img',
+    type: 'src',
+    attributes: ['src']
+  },
+  {
+    tag: 'img',
+    type: 'srcset',
+    attributes: ['srcset']
+  },
+  {
+    tag: 'input',
+    type: 'src',
+    attributes: ['src']
+  },
+  {
+    tag: 'object',
+    type: 'src',
+    attributes: ['data']
+  },
+  {
+    tag: 'source',
+    type: 'src',
+    attributes: ['src']
+  },
+  {
+    tag: 'source',
+    type: 'srcset',
+    attributes: ['srcset']
+  },
+  {
+    tag: 'track',
+    type: 'src',
+    attributes: ['src']
+  },
+  {
+    tag: 'video',
+    type: 'src',
+    attributes: ['poster', 'src']
+  },
+  {
+    tag: 'image',
+    type: 'src',
+    attributes: ['href', 'xlink:href']
+  },
+  {
+    tag: 'use',
+    type: 'src',
+    attributes: ['href', 'xlink:href']
+  },
+  {
+    tag: 'link',
+    type: 'src',
+    attributes: ['href'],
+    filter: (attribute, value, node) => {
+      const relProp = findProp(node, 'rel')
+      if (
+        relProp &&
+        relProp.type === NodeTypes.ATTRIBUTE &&
+        relProp.value?.content &&
+        [
+          'stylesheet',
+          'icon',
+          'shortcut icon',
+          'mask-icon',
+          'apple-touch-icon',
+          'apple-touch-icon-precomposed',
+          'apple-touch-startup-image',
+          'manifest',
+          'prefetch',
+          'preload'
+        ].includes(relProp.value.content)
+      ) {
+        return true
+      }
+
+      const itempropProp = findProp(node, 'itemprop')
+      if (
+        itempropProp &&
+        itempropProp.type === NodeTypes.ATTRIBUTE &&
+        itempropProp.value?.content &&
+        [
+          'image',
+          'logo',
+          'screenshot',
+          'thumbnailurl',
+          'contenturl',
+          'downloadurl',
+          'duringmedia',
+          'embedurl',
+          'installurl',
+          'layoutimage '
```

Comment:
a typo 'layoutimage'

Reply:
Gosh My apologies, I think I corrected that one around ten times while messing around with git.